### ISSUE TITLE
Is the objective function of DANN defined incorrectly in the paper?

### ISSUE THREAD

**Issue by ugurcanakyuz** [fungtion/DANN#21]
The loss function in the paper, below, consists of a negative lambda (λ) value for domain loss. However, in your implementation, at this [line](https://github.com/fungtion/DANN/blob/476147f70bb818a63bb3461a6ecc12f97f7ab15e/train/main.py#L148), you sum all the loss values instead. Isn't this opposite of the loss defined in the paper?

![image](https://github.com/user-attachments/assets/770ad46f-cdee-4818-b358-a5a39408c73c)

But, in my opinion, summing up all the loss values as you did is correct because while the domain head tries to distinguish the domain and minimize its loss, the feature extractor tries to confuse it and maximizes its loss with the help of updates based on the GRL layer. If the negative lambda value was used in the loss, it would also confuse the domain head, and training would be unstable.

 So, is the negative lambda value in the paper wrong, or am I missing something?

**Comment by ugurcanakyuz** [comment]
I was wrong :). The loss function was defined with this formula: 

![Screenshot 2024-08-14 111606](https://github.com/user-attachments/assets/9d426995-25ea-417c-99cd-b7d43622d325)

"We can then define the objective “pseudo-function” of (θf , θy , θd) that is being optimized by the stochastic gradient descent within our method:"


Yet, they define it with this formula in the Journal [paper](https://jmlr.org/papers/volume17/15-239/15-239.pdf):
![image](https://github.com/user-attachments/assets/a3dedd33-9c36-4418-9072-e113d7305d09)


I first started to read the journal paper. That made me confused.
